wandb/sdk/artifacts/artifact.py

```diff
@@ -261,9 +261,23 @@ def _from_name(
         if is_artifact_registry_project(project):
             try:
                 entity = InternalApi()._resolve_org_entity_name(entity, organization)
-            except ValueError as e:
-                wandb.termerror(str(e))
-                raise
+            except ValueError as entity_error:
+                if organization and organization != entity:
+                    try:
+                        entity = InternalApi()._resolve_org_entity_name(
+                            organization, organization
+                        )
+                    except ValueError as org_error:
+                        wandb.termerror(
+                            f"Error resolving organization of entity: {entity!r}, failed with error: {entity_error!r}."
+                        )
+                        wandb.termerror(
+                            f"Defaulted to trying to use {organization!r} as an org entity and failed with error: {org_error!r}."
+                        )
+                        raise
+                else:
+                    wandb.termerror(str(entity_error))
+                    raise
```

Comment:
Can we do something like this instead?

```suggestion
           try:
               entity = InternalApi()._resolve_org_entity_name(entity, organization)
               if organization and organization != entity:
                   entity = InternalApi()._resolve_org_entity_name(
                            organization, organization
                        )
            except ValueError as entity_error:
                wandb.termerror(
                    f"Error resolving organization of entity: {entity!r}, failed with error: {entity_error!r}."
                    )
                    raise                       
            except ValueError as org_error:
                wandb.termerror(
                    f"Defaulted to trying to use {organization!r} as an org entity and failed with error: {org_error!r}."
                )
                    raise       
```

Reply:
hm my IDE, says the second `except ValueError as org_error:` is unreachable

Reply:
oh sorry thats cuz they're both `ValueError`s

Reply:
maybe instead of nesting them, you could just separate the two and make the two try-except blocks sequential?

Reply:
but i don't want it to error on the first try except. not sure if having 2 trys and then a catch make sense

Reply:
I think maybe this is cleaner?

```
try:
    entity = InternalApi()._resolve_org_entity_name(entity, organization)
except ValueError as entity_error:
    if not organization or organization == entity:
        wandb.termerror(str(entity_error))
        raise

    # Try to resolve using the organization as the entity
    try:
        entity = InternalApi()._resolve_org_entity_name(organization, organization)
    except ValueError as org_error:
        wandb.termerror(
            f"Error resolving entity: {entity!r}, failed with error: {entity_error!r}."
        )
        wandb.termerror(
            f"Attempted to use {organization!r} as the organization entity and failed with error: {org_error!r}."
        )
        raise
```

Reply:
Makes sense to me now. Sorry about the back and forth!